src/app/clients/getClientsTableData.ts

```diff
@@ -0,0 +1,152 @@
+import supabase, { Schema } from "@/supabase";
+import { Datum } from "@/components/Tables/Table";
+
+export interface ClientsTableRow extends Datum {
+    parcelId: Schema["parcels"]["primary_key"];
+
+    flaggedForAttention: boolean;
+    requiresFollowUpPhoneCall: boolean;
+
+    fullName: Schema["clients"]["full_name"];
+    familyCategory: string;
+    addressPostcode: Schema["clients"]["address_postcode"];
+
+    collectionCentre: string;
+    congestionChargeApplies: boolean;
+
+    packingTimeLabel: string;
+    lastStatus: string;
+}
+
+export type ProcessingData = Awaited<ReturnType<typeof getProcessingData>>;
+
+// eslint-disable-next-line @typescript-eslint/explicit-function-return-type
```

Comment:
I believe that the best way for me to obtain the return type of the custom `supabase` query is to define the type through type inference from the request function itself - that's what the line above the `eslint-disable` is for.

The `eslint-disable` is required to ignore the lack of an explicit return type on the function.

The type `ProcessingData` can't be derived from the inferred return type of the function _and_ be used to explicitly annotate the return type as its definition then becomes circular.

Reply:
I see I see. Might have to check with Tom if he likes it or not, because in general disabling eslint isn't a good sign (?) but I get what you mean, as the joins you are doing here is quite complex (fyi I did a join for calendar but that is literally just for the client full_name so it is rather easy for me to define the type myself)